matter_server/client/models/node.py

```diff
@@ -0,0 +1,253 @@
+            self.endpoints[endpoint_id] = MatterEndpoint(
+                endpoint_id=endpoint_id, attributes_data=attributes_data, node=self
+            )
+        # lookup device types from node data
+        for endpoint in self.endpoints.values():
+            # get DeviceTypeList Attribute on the Descriptor cluster
+            attribute = endpoint.get_attribute(
+                Clusters.Descriptor, Clusters.Descriptor.Attributes.DeviceTypeList
+            )
+            if not attribute:
+                continue
+
+            for dev_info in attribute.value:
+                if not isinstance(
+                    dev_info, Clusters.Descriptor.Structs.DeviceTypeStruct
+                ):
+                    dev_info = Clusters.Descriptor.Structs.DeviceTypeStruct(**dev_info)
+
+                device_type = DEVICE_TYPES.get(dev_info.type)
+                if device_type is None:
+                    LOGGER.debug("Found unknown device type %s", dev_info)
+                    continue
+
+                instance: MatterDeviceTypeInstance[Any] = MatterDeviceTypeInstance(
+                    self, device_type, endpoint.endpoint_id, dev_info.revision
+                )
+                if device_type is RootNode:
+                    self.root_device_type_instance = instance
+                elif device_type is Aggregator:
+                    self.aggregator_device_type_instance = instance
+                else:
+                    self.device_type_instances.append(instance)
+
+        if self.root_device_type_instance is None:
+            raise ValueError("No root device found")
+
+        # parse node devices
+        self.node_devices = []
+        if self.aggregator_device_type_instance:
+            for instance in self.device_type_instances:
+                if instance.device_type == BridgedDevice:
+                    self.node_devices.append(MatterBridgedNodeDevice(instance))
+        else:
+            self.node_devices.append(MatterNodeDevice(self))
+
+    @property
+    def node_id(self) -> int:
+        """Return Node ID."""
+        return self.node_data.node_id
+
+    @property
+    def available(self) -> bool:
+        """Return availability of the node."""
+        return self.node_data.available
+
+    def get_attribute_value(
+        self,
+        endpoint: int,
+        cluster: type[_CLUSTER_T] | int,
+        attribute: str | int | type[_ATTRIBUTE_T],
+    ) -> Any:
+        """Return Matter Cluster Attribute value for given parameters."""
+        return self.endpoints[endpoint].get_attribute_value(cluster, attribute)
+
+    def has_cluster(
+        self, cluster: type[_CLUSTER_T] | int, endpoint: int | None = None
+    ) -> bool:
+        """Check if node has a specific cluster on any of the endpoints."""
+        return any(
+            x
+            for x in self.endpoints.values()
+            if x.has_cluster(cluster)
+            and (x.endpoint_id == endpoint or endpoint is None)
```

Comment:
Start with the None check. Then we can save one attribute access if the check short-circuits.